lang/lang.go

```diff
@@ -0,0 +1,185 @@
+	// More info available on the `LocalizePluralKey` function.
+	XN = LocalizePluralKey
+
+	bundle    *i18n.Bundle
+	localizer *i18n.Localizer
+
+	//go:embed translations
+	translations embed.FS
+	translated   []language.Tag
+)
+
+// Localize asks the translation engine to translate a string, this behaves like the gettext "_" function.
+// The string can be templated and the template data can be passed as a struct with exported fields,
+// or as a map of string keys to any suitable value.
+func Localize(in string, data ...any) string {
+	return LocalizeKey(in, in, data...)
+}
+
+// LocalizeKey asks the translation engine for the translation with specific ID.
+// If it cannot be found then the fallback will be used.
+// The string can be templated and the template data can be passed as a struct with exported fields,
+// or as a map of string keys to any suitable value.
+func LocalizeKey(key, fallback string, data ...any) string {
+	var d0 any
+	if len(data) > 0 {
+		d0 = data[0]
+	}
+
+	ret, err := localizer.Localize(&i18n.LocalizeConfig{
+		DefaultMessage: &i18n.Message{
+			ID:    key,
+			Other: fallback,
+		},
+		TemplateData: d0,
+	})
+
+	if err != nil {
+		fyne.LogError("Translation failure", err)
+		return fallbackWithData(key, fallback, d0)
+	}
+	return ret
+}
+
+// LocalizePlural asks the translation engine to translate a string from one of a number of plural forms.
+// This behaves like the ngettext function, with the `count` parameter determining the plurality looked up.
+// The string can be templated and the template data can be passed as a struct with exported fields,
+// or as a map of string keys to any suitable value.
+func LocalizePlural(in string, count int, data ...any) string {
+	return LocalizePluralKey(in, in, count, data...)
+}
+
+// LocalizePluralKey asks the translation engine for the translation with specific ID in plural form.
+// This behaves like the npgettext function, with the `count` parameter determining the plurality looked up.
+// If it cannot be found then the fallback will be used.
+// The string can be templated and the template data can be passed as a struct with exported fields,
+// or as a map of string keys to any suitable value.
+func LocalizePluralKey(key, fallback string, count int, data ...any) string {
+	var d0 any
+	if len(data) > 0 {
+		d0 = data[0]
+	}
+
+	ret, err := localizer.Localize(&i18n.LocalizeConfig{
+		DefaultMessage: &i18n.Message{
+			ID:    key,
+			Other: fallback,
+		},
+		PluralCount:  count,
+		TemplateData: d0,
+	})
+
+	if err != nil {
+		fyne.LogError("Translation failure", err)
+		return fallbackWithData(key, fallback, d0)
+	}
+	return ret
+}
+
+// AddTranslations allows an app to load a bundle of translations.
+// The language that this relates to will be inferred from the resource name, for example "fr.json".
+// The data should be in json format.
+func AddTranslations(r fyne.Resource) error {
```

Comment:
Just adding a comment here that we discussed on the call about if resource is the right type here? With `embed` being more and more popular it might be a bit annoying to wrap it into a resource (and I and maybe more people mostly think of resources as images). Thoughts?

Reply:
I think we agreed to add embed.FS support as an additional option - as we need the file name and file content to support the features, which is where the translations for locale method comes in.
I can add an FS helper though.

Reply:
Added